lib/StaticNarrative/exporter/data_exporter.py

```diff
@@ -55,32 +57,46 @@ def export_narrative_data(
     ws_data = ows.list_objects_with_sets(ws_id=wsid, include_metadata=1)
 
     filtered_data = []
+    indexed_data = {}
     type_info = {}
     for item in ws_data:
         obj = item["object_info"]
         obj_type = obj[2].split("-")[0]
         if obj_type in IGNORED_TYPES:
             continue
+        obj_upa = generate_upa(item["object_info"])
```

Comment:
shortcut for that annoying `f"{item[4]}/{item[0]}/{item[6]}" thing

Reply:
We might seriously consider making a kbase_upa python package or something. All those helper functions are some project-wide copypasta. It'd be nice to just pip install it.

Reply:
Yes! The NarrativeRef class and these various upa-related functions. How many projects have I added that regex for detecting a KBase object identifier to?!